src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -326,6 +330,28 @@ internal virtual NamedTypeSymbol LookupMetadataType(ref MetadataTypeName emitted
             }
 
 Done:
+            if (isTopLevel && scope is not PENamespaceSymbol && GeneratedNameParser.TryParseFileTypeName(emittedTypeName.UnmangledTypeName, out string? displayFileName, out int ordinal, out string? sourceName))
```

Comment:
> &&  [](http://example.com/codeflow?start=26&length=4)

Minor: Consider splitting the `if` condition across multiple lines at the `&&`. (A couple of times, I've found the current single line challenging to read.)